pkg/controller/elasticsearch/nodespec/statefulset.go

```diff
@@ -161,6 +162,16 @@ func setVolumeClaimsControllerReference(
 			continue
 		}
 
+		// TODO we can skip the whole setting of ownerReferences for the retain case if we can make sure via validation
```

Comment:
I'm wondering if the whole ownerRef on volume claim trick is not a mistake to start with.
As in: we could stop doing that (but keep existing ownerrefs for bwc), and gc pvcs "manually" instead.
By doing so users would be free to change their delete policy at any time (unless they rely on the bwc mode).

Reply:
So I have been thinking about that as well. The suggestion are IMO two different implementation variants: 

* we would not use the template mechanism of the stateful set to set `ownerReferences`, but instead set them directly on the PVC. This makes the creating of the Pod and its volume a multi step process with more potential for failure (what if we fail to set/remove the `ownerReference`).  The main advantage would be that users would not be constrained to chose one of the policies at cluster creation and could change their mind afterwards. 

* the other suggestion here is to manually garbage collect PVCs. We would need to annotate the PVCs with the current deletion policy so that we know what to garbage collect after the corresponding cluster has been deleted. This of course means we cannot use the sset template mechanism to place these annotations if we want to allow users to change the policy (same as when setting the `owerReference` directly). It therefore has similar failure cases (failing to set or update the annotation before garbage collection kicks in) The arguments in favour of using our own garbage collection that I can see are only that we are already doing a limited form of garbage collection during scale down of individual nodes and we could therefore standardize on one mechanism. The arguments against this are that `ownerReference`s are already implementing exactly this feature which we would duplicate. 

We could also add the 'going behind the sset's back' approach as a follow up PR to this work as it would be API compatible and we would only lift restrictions not introduce new ones.

Reply:
We discussed out of band and I will change this PR to set the `ownerReferences` without relying on the templating in the StatefulSets. This will free us to change policy as the user desires. I need to verify that this will not cause problems if the effective `ownerReference` in the PVC contradicts the volumeClaimTemplate e.g. if the StatefulSet controller will try to reconcile the `ownerReference`. If that turns out to be the case we would introduce an exception for existing clusters which could not change their `VolumeClaimDeletePolicy`